esrally/driver/runner.py

```diff
@@ -404,7 +404,7 @@ async def __call__(self, *args):
                     props = parse(return_value, assertion["property"])
                     self.check_assertion(op_name, assertion, props, props_expand_keys=False)
             else:
-                self.logger.debug("Skipping assertion check in [%s] as [%s] does not return a dict.", op_name, repr(self.delegate))
+                raise exceptions.DataError(f"Cannot check assertion in [{op_name}] as [{repr(self.delegate)}] does not return a dict.")
```

Comment:
The only nit I have here is that perhaps this could say `did not return a dict`, as it could be that the operation *typically* returns a valid parseable response, but in this instance returned a stacktrace or something

Reply:
Thanks, I fixed that. Well it could also return JSON encoded as bytes in a BytesIO but I did not want to overcomplicate the error message.